frontend/src/metabase/lib/settings.ts

```diff
@@ -90,58 +99,95 @@ class MetabaseSettings {
     });
   }
 
+  /**
+   * @deprecated call appropriate actions when modifying the setting
+   */
   on(key: SettingKey, callback: SettingListener) {
     this._listeners[key] = this._listeners[key] || [];
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     this._listeners[key]!.push(callback);
   }
 
-  // these are all special accessors which provide a lookup of a property plus some additional help
+  /**
+   * @deprecated use getSetting(state, "admin-email")
+   */
   adminEmail() {
     return this.get("admin-email");
   }
 
+  /**
+   * @deprecated use getSetting(state, "enable-enhancements?")
+   */
   enhancementsEnabled() {
     return this.get("enable-enhancements?");
   }
 
+  /**
+   * @deprecated use getSetting(state, "email-configured?")
+   */
   isEmailConfigured(): boolean {
     return !!this.get("email-configured?");
   }
 
+  /**
+   * @deprecated use getSetting(state, "is-hosted?")
+   */
   isHosted(): boolean {
     return !!this.get("is-hosted?");
   }
 
+  /**
+   * @deprecated use getSetting(state, "cloud-gateway-ips")
+   */
   cloudGatewayIps(): string[] {
     return this.get("cloud-gateway-ips") || [];
   }
 
+  /**
+   * @deprecated use getSetting(state, "has-user-setup")
+   */
   hasUserSetup() {
     return this.get("has-user-setup");
   }
 
+  /**
+   * @deprecated use getSetting(state, "hide-embed-branding?")
+   */
   hideEmbedBranding() {
     return this.get("hide-embed-branding?");
   }
 
+  /**
+   * @deprecated use getSetting(state, "google-auth-enabled")
+   */
   isGoogleAuthEnabled() {
     return this.get("google-auth-enabled");
   }
 
+  /**
+   * @deprecated use getSetting(state, "ldap-enabled")
+   */
   isLdapEnabled() {
     return this.get("ldap-enabled");
   }
 
+  /**
+   * @deprecated use getSetting(state, "ldap-configured?")
+   */
   isLdapConfigured() {
     return this.get("ldap-configured?");
   }
 
-  // JWT or SAML is enabled
+  /**
+   * @deprecated use getSetting(state, "other-sso-enabled?")
+   */
   isOtherSsoEnabled() {
     return this.get("other-sso-enabled?");
   }
 
+  /**
+   * @deprecated use getSetting(state, ...)
```

Comment:
For a case like this, would it be worth creating a new selector and putting it in `frontend/src/metabase/selectors/settings.ts`, and then mentioning to use that selector instead e.g. `getIsSsoEnabled`?

It'd be unclear what's the suggested alternative in this case.